### PR TITLE
[#9] [#11] [UI] As a user, I can see the iOS 10 Screen with Dynamic Custom Font #9

### LINKED ISSUE #9: [UI] As a user, I can see the iOS 10 Screen with Dynamic Custom Font
## Why

To display dynamic custom font with iOS 10 limitation.

## Acceptance Criteria

Show a screen with:
- Image view
- Title "iOS 10".
- Bold title text "Font Name"
- Text "iOS Version"
- Text "UIKit"
- Caption text "A demo from NimbleHQ"

Screen should be in vertical scroll view to support multiple screen sizes.

## Design

iOS10

### LINKED ISSUE #11: [UI] As a user, I can see the iOS 10 Screen with Override Settings
## Why

To override iOS 10 settings for font size.

## Acceptance Criteria

The iOS 10 screen should include:
- Image and text section.
- Label "Override System Font Size".
- On/Off Toggle
- Label "Override Font Size".
- Slider for the font size.
- Font size indicator. 

## Design

iOS10

### PR DESCRIPTION
**Pull request by blyscuit**
close #9 
close #11 

## What happened

Show a screen with:

Image view
Title "iOS 10".
Bold title text "Font Name"
Text "iOS Version"
Text "UIKit"
Caption text "A demo from NimbleHQ"
Image and text section.
Label "Override System Font Size".
On/Off Toggle
Label "Override Font Size".
Slider for the font size.
Font size indicator.
Screen should be in vertical scroll view to support multiple screen sizes.
 
## Insight

Move part of iOS11UIKitViewController to a UIView Class for reuse.

ViewModel stays the same hence the data showing incorrectly.
 
## Proof Of Work
.

### CODE REVIEW

File: CustomDynamicFont/Sources/Presentation/Modules/iOS10/IOS10ViewController.swift [hunk 1]
```diff
@@ -6,6 +6,45 @@
...
+    @Injected private var viewModel: UIKitViewModelProtocol
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpLayout()
+        bindViewModel()
+    }
+
+    override func updateFonts(notification _: Notification) {
+        print("updating font")
```
Comment (minhnimble) [inline CustomDynamicFont/Sources/Presentation/Modules/iOS10/IOS10ViewController.swift:27]: We can add a TODO here as well

File: CustomDynamicFont/Sources/Presentation/Modules/iOS10/DynamicFontController.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+extension DynamicFontController where Self: UIViewController {
+
+    func setUpContentSizeNotification() {
+        NotificationCenter.default.addObserver(
+            forName: UIContentSizeCategory.didChangeNotification,
+            object: nil,
+            queue: .main
+        ) { [weak self] notification in
+            self?.updateFonts(notification: notification)
+        }
```
Comment (minhnimble) [inline CustomDynamicFont/Sources/Presentation/Modules/iOS10/DynamicFontController.swift]: Just curious, can we use RxSwift for notification center instead?

Comment (blyscuit) [reply CustomDynamicFont/Sources/Presentation/Modules/iOS10/DynamicFontController.swift]: Good idea.

**Review by minhnimble — APPROVED** [review]

**Comment by minhnimble** [comment]
Code-wise LGTM. Please fix the conflicts 🛠🦀

File: CustomDynamicFont/Resources/Localizations/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -10,6 +10,14 @@
...
 "ios11uikit.lifecyclelabel.title" = "UIKit";
+"ios12swiftui.lifecyclelabel.title" = "SwiftUI";
+"ios11uikit.overrideFontTitle.title" = "Override System Font Size";
+"ios11uikit.overrideFontSegmentOptions.on" = "On";
+"ios11uikit.overrideFontSegmentOptions.off" = "Off";
+"ios11uikit.overrideFontScaleTitle.title" = "Override Font Size";
+"overridefontsize.small.label" = "SM";
+"overridefontsize.normal.label" = "MD";
+"overridefontsize.large.label" = "LG";
+"overridefontsize.extraLarge.label" = "XL";
```
Comment (markgravity) [inline CustomDynamicFont/Resources/Localizations/en.lproj/Localizable.strings]: I think these should be in camel case? 🤔 
```
ios12swiftui -> ios12SwiftUI
overridefontsize -> overrideFontSize
```

**Review by minhnimble — CHANGES_REQUESTED** [review]
Your CI is failed, please help check and fix it 🛠

**Review by markgravity — APPROVED** [review]
LGTM

**Review by minhnimble — APPROVED** [review]

**Pull request merged**